app/src/main/java/ch/epfl/favo/view/tabs/favorList/FavorViewHolder.java

```diff
@@ -57,18 +56,26 @@ void bind(Context context, @NonNull Favor favor, View parentView, IFavorViewMode
   @RequiresApi(api = Build.VERSION_CODES.N)
   private void setItemDisplay(Favor favor, Context context) {
     mTitleView.setText(favor.getTitle());
-    mRewardView.setText(context.getString(R.string.favo_coins_item_placeholder, favor.getReward()));
+    mRewardView.setText(context.getString(R.string.num_coins, (int)favor.getReward()));
 
-    if (currentUser.getUid().equals(favor.getRequesterId())) {
-      if (currentUser.getPhotoUrl() != null) {
-        Glide.with(context).load(currentUser.getPhotoUrl()).into(mRequesterIconView);
-      }
+    if (currentUser.getId().equals(favor.getRequesterId())) {
       mRequesterView.setText(
-          context.getString(R.string.user_name_item_placeholder, currentUser.getDisplayName()));
+          context.getString(R.string.user_name_item_placeholder, currentUser.getName()));
+      if (currentUser.getProfilePictureUrl() != null) {
+        Glide.with(context).load(currentUser.getProfilePictureUrl()).into(mRequesterIconView);
+      }
     } else {
       UserUtil.getSingleInstance()
           .findUser(favor.getRequesterId())
-          .thenAccept(user -> mRequesterView.setText(getUserName(user)));
+          .thenAccept(
+              user -> {
+                mRequesterView.setText(getUserName(user));
+                if (user.getProfilePictureUrl() != null) {
+                  Glide.with(context)
```

Comment:
Here we can maybe use one of the Cache options of Glide to prevent multiple reloading

Reply:
Can you supply the code below?